src/core.py

```diff
@@ -59,7 +67,7 @@ def on_config(self, config):
         for key in theme_override:
             if key not in config["theme"]:
```

Comment:
I noticed that, because `config["theme"]` is always an instance of `Theme`, any properties that are set by default (e.g. `language`) can never be overridden.  You'll likely want to tweak this logic to make this works as expected.